ddtrace/contrib/redis/__init__.py

```diff
@@ -16,14 +16,92 @@
     # Use a pin to specify metadata related to this client
     Pin.override(client, service='redis-queue')
 """
+from wrapt import wrap_function_wrapper as _w
 
-from ...utils.importlib import require_modules
+from ...ext import AppTypes, redis as redisx
+from ...pin import Pin
+from ...utils.wrappers import unwrap
+from ...utils.install import install_module_import_hook
+from .tracers import get_traced_redis
+from .util import format_command_args, _extract_conn_tags
 
-required_modules = ['redis', 'redis.client']
 
-with require_modules(required_modules) as missing_modules:
-    if not missing_modules:
-        from .patch import patch
-        from .tracers import get_traced_redis, get_traced_redis_from
+__all__ = [
+    'get_traced_redis',
+    'patch',
+]
 
-        __all__ = ['get_traced_redis', 'get_traced_redis_from', 'patch']
+
+def _patch_redis_client(redisclient):
+    _w('redis.client', 'BasePipeline.execute', traced_execute_pipeline)
```

Comment:
are we able to do `_w(redisclient.BasePipeline, 'execute', traced_execute_pipeline)` ?

Reply:
it would be nice to be able to interact with the module object directly since we have it in this function anyways (as opposed to having `wrapt` just pull it from `sys.modules`.

this might also help side-skirt some potential issues with how `wrapt` traverses attributes.

Reply:
I really like this idea. 👍